### PR TITLE
change treeView appearance

### PR DESCRIPTION
**Pull request by anistouri**

### CODE REVIEW

File: src/components/react-hook-form/utils/submit-button.js [hunk 1]
```diff
@@ -10,11 +10,15 @@ import { Button } from '@mui/material';
 import { useFormState } from 'react-hook-form';
 import { FormattedMessage } from 'react-intl';
 
-const SubmitButton = ({ onClick, disabled = false }) => {
+const SubmitButton = ({ onClick, disabled = false, variant }) => {
```
Comment (sBouzols) [inline src/components/react-hook-form/utils/submit-button.js:13]:
```suggestion
const SubmitButton = ({ ...buttonProps }) => {
```

File: src/components/react-hook-form/utils/submit-button.js [hunk 2]
```diff
@@ -10,11 +10,15 @@ import { Button } from '@mui/material';
...
     const { isDirty } = useFormState();
 
     return (
-        <Button onClick={onClick} disabled={!isDirty || disabled}>
+        <Button
+            onClick={onClick}
+            disabled={!isDirty || disabled}
+            variant={variant}
+        >
```
Comment (sBouzols) [inline src/components/react-hook-form/utils/submit-button.js:21]:
```suggestion
        <Button
            ...buttonProps,
            disabled={!isDirty || (buttonProps.disabled ?? false)}
        >
```

File: src/components/react-hook-form/utils/submit-button.js (hunk 1 above)
Comment (anistouri) [reply src/components/react-hook-form/utils/submit-button.js]: Done

File: src/components/react-hook-form/utils/submit-button.js (hunk 2 above)
Comment (anistouri) [reply src/components/react-hook-form/utils/submit-button.js]: Done

File: src/components/react-hook-form/utils/cancel-button.js [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
             <FormattedMessage id="cancel" />
         </Button>
     );
 };
 
 CancelButton.propTypes = {
-    onClick: PropTypes.func.isRequired,
-    variant: PropTypes.string,
-    disabled: PropTypes.bool,
     withCustomColor: PropTypes.bool,
```
Comment (sBouzols) [inline src/components/react-hook-form/utils/cancel-button.js:29]: remove

File: src/components/react-hook-form/utils/cancel-button.js [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
+    }, [withCustomColor, buttonProps?.color]);
     return (
-        <Button
-            onClick={onClick}
-            variant={variant}
-            disabled={disabled}
-            color={withCustomColor ? 'customButton' : 'primary'}
-        >
+        <Button {...buttonProps} color={colorKey}>
             <FormattedMessage id="cancel" />
```
Comment (sBouzols) [inline src/components/react-hook-form/utils/cancel-button.js:23]: add a translation file for this key for this component `cancel-button-fr.js` `cancel-button-en.js`

File: src/components/react-hook-form/utils/submit-button.js [hunk 5]
```diff
@@ -9,15 +9,23 @@ import React from 'react';
...
+const SubmitButton = ({ ...buttonProps }) => {
     const { isDirty } = useFormState();
 
     return (
-        <Button onClick={onClick} disabled={!isDirty || disabled}>
+        <Button
+            {...buttonProps}
+            disabled={!isDirty || (buttonProps?.disabled ?? false)}
+        >
             <FormattedMessage id="validate" />
```
Comment (sBouzols) [inline src/components/react-hook-form/utils/submit-button.js:22]: add translation files for this too

File: src/components/TreeViewFinder/TreeViewFinder.js [hunk 6]
```diff
@@ -114,6 +115,7 @@ const TreeViewFinder = (props) => {
         multiselect,
         sortMethod,
         className,
+        withCustomColor = true,
```
Comment (sBouzols) [inline src/components/TreeViewFinder/TreeViewFinder.js:118]: `CancelButtonProps`

File: src/components/TreeViewFinder/TreeViewFinder.js [hunk 7]
```diff
@@ -344,17 +346,17 @@ const TreeViewFinder = (props) => {
             </DialogContent>
             <DialogActions>
                 <Button
```
Comment (sBouzols) [inline src/components/TreeViewFinder/TreeViewFinder.js:348]: Use the CancelButton component here

**Review by sBouzols — CHANGES_REQUESTED** [review]
Some changes to be more generic

File: src/components/react-hook-form/utils/cancel-button.js [hunk 8]
```diff
@@ -1,31 +1,33 @@
...
+        }
+    }, [withCustomColor, buttonProps?.color]);
     return (
-        <Button
-            onClick={onClick}
-            variant={variant}
-            disabled={disabled}
-            color={withCustomColor ? 'customButton' : 'primary'}
-        >
+        <Button {...buttonProps} color={colorKey}>
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/components/react-hook-form/utils/cancel-button.js:22]: not the good syntax, should do something else in this style

[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/react-hook-form/utils/cancel-button.js]: The color prop has been removed. It is now managed by the parent component

File: src/components/react-hook-form/utils/cancel-button.js (hunk 3 above)
Comment (anistouri) [reply src/components/react-hook-form/utils/cancel-button.js]: done

File: src/components/react-hook-form/utils/cancel-button.js (hunk 4 above)
Comment (anistouri) [reply src/components/react-hook-form/utils/cancel-button.js:16]: done

File: src/components/react-hook-form/utils/submit-button.js (hunk 5 above)
Comment (anistouri) [reply src/components/react-hook-form/utils/submit-button.js:22]: done

File: src/components/TreeViewFinder/TreeViewFinder.js [hunk 9]
```diff
@@ -344,17 +346,17 @@ const TreeViewFinder = (props) => {
...
-                    variant="contained"
                     style={{ float: 'left', margin: '5px' }}
                     onClick={() => {
                         onClose([]);
                         setSelected([]);
                     }}
+                    color={withCustomColor ? 'customButton' : 'primary'}
```
[the remark this answers is not in the document]
Comment (anistouri) [reply src/components/TreeViewFinder/TreeViewFinder.js]: done

File: src/components/TreeViewFinder/TreeViewFinder.js (hunk 6 above)
Comment (anistouri) [reply src/components/TreeViewFinder/TreeViewFinder.js]: done

File: src/components/TreeViewFinder/TreeViewFinder.js (hunk 7 above)
Comment (sBouzols) [reply src/components/TreeViewFinder/TreeViewFinder.js:348]: well we could do it later

Comment (anistouri) [reply src/components/TreeViewFinder/TreeViewFinder.js]: done

**Review by sBouzols — APPROVED** [review]

**Review by YenguiSeddik — APPROVED** [review]

**Pull request merged**